integration-tests/tests/hacbs/add-component-to-application-via-hacbs-tab.spec.ts

```diff
@@ -0,0 +1,116 @@
+    const applicationDetailPage = new ApplicationDetailPage();
+    const publicRepoOfJava = 'https://github.com/dheerajodha/devfile-sample-code-with-quarkus';
+    const publicRepoOfGo = 'https://github.com/devfile-samples/devfile-sample-go-basic.git';
+    const publicRepoOfNodeJS = 'https://github.com/nodeshift-starters/devfile-sample.git';
+    const componentOfJava = 'java-quarkus';
+    const componentOfGo = 'go';
+    const componentOfNodeJS = 'nodejs'
+
+    before(function () {
+        // Enable HACBS
+        Common.enableHACBS();
+    });
+
+    after(function () {
+        //Open components page and delete all 3 components
+        applicationDetailPage.deleteComponent(componentOfJava);
+        applicationDetailPage.createdComponentNotExists(componentOfJava);
+
+        applicationDetailPage.deleteComponent(componentOfGo);
+        applicationDetailPage.createdComponentNotExists(componentOfGo);
+
+        applicationDetailPage.deleteComponent(componentOfNodeJS);
+        applicationDetailPage.createdComponentNotExists(componentOfNodeJS);
+
+        Applications.deleteApplication(applicationName);
+    });
+
+    describe('Create an Application with a component', () => {
+        it('Set Application Name', () => {
+            Applications.createApplication(applicationName);
+        })
+
+        it('Validate Repo', () => {
+            // Enter git repo URL
+            addComponent.setSource(publicRepoOfJava);
+            // Check if the source is validated
+            addComponent.waitRepoValidated();
+        });
+
+        it('Setup Git Options', () => {
+            addComponent.clickGitOptions();
+
+            //Next block commented out because of bug:
+            //https://issues.redhat.com/browse/HAC-1285
+
+            //addComponent.setGitReference(gitReference);
+            //addComponent.setContextDir(contextDir);
+            addComponent.clickNext();
+        });
+
+        it('Create Application', () => {
+            componentPage.createApplication();
+            applicationDetailPage.createdComponentExists(componentOfJava, applicationName);
+        });
+    })
+
+    describe('Add a new component using the "Components" tab', () => {
+        it("Use HACBS 'Components' tabs to start adding a new compoent", () => {
+            //Open components page
+            overviewTabPage.goTocomponentsTab();
+            componentsTabPage.clickAddComponent();
+        });
+
+        it('Validate Repo', () => {
+            // Enter git repo URL
+            addComponent.setSource(publicRepoOfGo);
+            // Check if the source is validated
+            addComponent.waitRepoValidated();
+            addComponent.clickNext();
+        });
+
+        it('Create Component and verify its visibility', () => {
+            componentPage.createApplication();
+            overviewTabPage.goTocomponentsTab();
+            applicationDetailPage.createdComponentExists(componentOfGo, applicationName);
+        });
```

Comment:
if these steps are being repeated with different values, it would be best to create a function to avoid code duplicities

Reply:
Done